drivers/adc/adc_nrfx_saadc.c

```diff
@@ -669,58 +650,50 @@ static int adc_nrfx_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
-static void saadc_irq_handler(const struct device *dev)
+static void event_handler(const nrfx_saadc_evt_t *event)
 {
-	if (nrf_saadc_event_check(NRF_SAADC, NRF_SAADC_EVENT_END)) {
-		nrf_saadc_event_clear(NRF_SAADC, NRF_SAADC_EVENT_END);
-
-		nrf_saadc_task_trigger(NRF_SAADC, NRF_SAADC_TASK_STOP);
+	nrfx_err_t err;
 
-#if defined(CONFIG_PM_DEVICE_RUNTIME)
-		pm_device_runtime_put(DEVICE_DT_INST_GET(0));
-#else
-		nrf_saadc_disable(NRF_SAADC);
-#endif
+	m_data.event = event;
```

Comment:
This pointer should not be stored. Instead, `event->data.done.p_buffer` should be supplied to `correct_single_ended()` as a parameter and only `event->data.done.size` should be stored for later use in `adc_context_update_buffer_pointer()`.